registry/registry.go

```diff
@@ -659,7 +659,18 @@ func ApprovePendingVersion(c *Space, pending *Version, app *App) (*Version, erro
 	channel := GetVersionChannel(release.Version)
 
 	// Cleaning the old versions
-	go CleanOldVersions(c, release.Slug, ChannelToStr(channel), conf.CleanNbMonths, conf.CleanNbMajorVersions, conf.CleanNbMinorVersions)
+	errs := make(chan error)
+	go func() {
+		err := CleanOldVersions(c, release.Slug, ChannelToStr(channel), conf.CleanNbMonths, conf.CleanNbMajorVersions, conf.CleanNbMinorVersions)
+		if err != nil {
+			errs <- err
+			return
+		}
+	}()
+
+	if err := <-errs; err != nil {
+		return nil, err
+	}
```

Comment:
It's not a good idea to use a channel for the error here. It means that we will wait that `CleanOldVersions` has finished before responding to the HTTP request (which was what we try to avoid by using a goroutine). In this case, we should just log the error, not return it.